### PR TITLE
Add Elevator and Intake Subsystems

### LINKED ISSUE #30: Implement Elevator Subsystem
**How does this improve the robot or repo?**
* Required to score in trap or amp

**Feature Description**
* See [Elevator.md](https://github.com/frc2609/Crescendo-2024-Planning/blob/main/Elevator.md)

### LINKED ISSUE #44: Elevator moves in one step, not one at a time!
**Bug Description**
* Elevator code & simulation is completely different from actual hardware. Assumed stage moves after intake reaches top, not simultaneously.

### PR DESCRIPTION
**Pull request by UserC2**
Fixes #30 
Fixes #44

Also adds Intake subsystem, see #47

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/Elevator.java
Comment (UserC2) [inline src/main/java/frc/robot/subsystems/Elevator.java:1]: From Greg:
* Don't use position/velocity conversions (do them yourself on the RIO) due to history of them not actually working.
* ^ make sure to scale upper/lower soft stops given to Spark MAX appropriately

Comment (newjanson) [reply src/main/java/frc/robot/subsystems/Elevator.java:1]: Make sure conversions are applied in getHeight() and setHeight()

**Comment by gregzetko** [comment]
Looks good! Ready for testing once hardware is ready

**Comment by UserC2** [comment]
See #44! Closed until fixed!

**Comment by UserC2** [comment]
Fixed

**Pull request merged**